flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/parser/ClientParser.java

```diff
@@ -0,0 +1,104 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client.cli.parser;
+
+import org.apache.flink.sql.parser.impl.FlinkSqlParserImplTokenManager;
+import org.apache.flink.sql.parser.impl.SimpleCharStream;
+import org.apache.flink.sql.parser.impl.Token;
+import org.apache.flink.table.client.gateway.SqlExecutionException;
+import org.apache.flink.table.operations.Operation;
+
+import javax.annotation.Nonnull;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** ClientParser use {@link FlinkSqlParserImplTokenManager} to do lexical analysis. */
+public class ClientParser implements SqlCommandParser {
+
+    /** A dumb implementation. TODO: remove this after unifying the SqlMultiLineParser. */
+    @Override
+    public Optional<Operation> parseCommand(String command) {
+        return Optional.empty();
+    }
+
+    public Optional<StatementType> parseStatement(@Nonnull String statement)
+            throws SqlExecutionException {
+        String trimmedStatement = statement.trim();
+        FlinkSqlParserImplTokenManager tokenManager =
+                new FlinkSqlParserImplTokenManager(
+                        new SimpleCharStream(new StringReader(trimmedStatement)));
+        List<Token> tokenList = new ArrayList<>();
+        Token token;
+        do {
+            token = tokenManager.getNextToken();
+            tokenList.add(token);
+        } while (token.endColumn != trimmedStatement.length());
+        return getStatementType(tokenList);
+    }
+
+    // ---------------------------------------------------------------------------------------------
+    private Optional<StatementType> getStatementType(List<Token> tokenList) {
+        Token firstToken = tokenList.get(0);
+
+        if (firstToken.kind == EOF || firstToken.kind == EMPTY || firstToken.kind == SEMICOLON) {
+            return Optional.empty();
+        }
+
+        if (firstToken.kind == IDENTIFIER) {
+            // unrecognized token
+            return getPotentialCommandType(firstToken.image);
+        } else if (firstToken.kind == EXPLAIN) {
+            return Optional.of(StatementType.EXPLAIN);
+        } else if (firstToken.kind == SHOW) {
+            return getPotentialShowCreateType(tokenList);
+        } else {
+            return Optional.of(StatementType.OTHER);
+        }
```

Comment:
When the statement is not ends with ;, it is not a statement. But it seems we will always return OTHER?

Reply:
To get a completed SQL statement is the function of the `SqlMultiLineParser`.  So I think we can make sure that statement here should be ended with ';'. And now I think all the test data should be ended with ';'.

Reply:
After checking the implementation of `SqlMultiLineParser`, here when the statement is incomplete, an SqlExecutionException should be thrown. I added the codes.